File: lib/domain/epub_reader/entities/epub_chapter.dart
import 'package:equatable/equatable.dart';

class EpubChapter extends Equatable {
  final String id;
  final String title;
  final String body;

  const EpubChapter({
    required this.id,
    required this.title,
    required this.body,
  });
  @override
  List<Object?> get props => [id, title, body];
}

File: android/app/src/main/kotlin/com/nmheeir/leafy/epub/models/EpubChapter.kt
package com.example.leafy.epub.models

import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.Serializable
import kotlinx.serialization.protobuf.ProtoNumber

/**
 * Represents a chapter in an epub book.
 *
 * @param absPath The absolute path of the chapter.
 * @param title The title of the chapter.
 * @param body The body of the chapter.
 */
@OptIn(ExperimentalSerializationApi::class)
@Serializable
data class EpubChapter @OptIn(ExperimentalSerializationApi::class) constructor(
    @ProtoNumber(1) val chapterId: String,
    @ProtoNumber(2) val absPath: String,
    @ProtoNumber(3) val title: String,
    @ProtoNumber(4) val body: String
)